### PR TITLE
Make Swift 5.2 the minimum required version

### PR DESCRIPTION
**Pull request by Davidde94**
We're applying this change across all of the NIO family of projects.

I've removed the docker files for earlier versions, and set the tools version. I've also added a brief note inside the contributing guide.

### CODE REVIEW

**Review by fabianfett — APPROVED** [review]
Awesome! Thank you!

**Comment by Davidde94** [comment]
Hey @yim-lee , would you be able to sort out the CI for this please? We need a few things:

- [ ] Remove 5.0 job
- [ ] Remove 5.1 job
- [ ] Update API breakage to be based on the Swift 5.4 Ubuntu 20.04 image
- [ ] Update the 5.2 job to use the new 16.04 docker file

**Review by Lukasa — APPROVED** [review]
LGTM.

**Comment by yim-lee** [comment]
@swift-server-bot test this please

**Review by yim-lee — COMMENTED** [review]
@Davidde94 CI changes are done. There is an issue with 1604.52.yaml that needs to be addressed but otherwise this should be good to go.

File: docker/docker-compose.1604.52.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  runtime-setup:
+    image: async-http-client:16.04-5.2
+    build:
+      args:
+        ubuntu_version: "xenial"
+        swift_version: "5.2"
+
+  test:
+    image: async-http-client:16.04-5.2
+    environment: []
```
[the remark this answers is not in the document]
Comment (Davidde94) [reply docker/docker-compose.1604.52.yaml:14]: Thanks!

**Pull request merged**